### PR TITLE
Convert typecheck to podutils job

### PR DESCRIPTION
**Pull request by cblecker**
Using what I learned from the godeps job, this rewrites the typecheck job using podutils

### CODE REVIEW

File: config/jobs/kubernetes-security/generated-security-jobs.yaml [hunk 1]
```diff
@@ -2393,28 +2389,25 @@ presubmits:
...
-        - --upload=gs://kubernetes-security-prow/pr-logs
-        - --timeout=75
-        - --scenario=execute
-        - --
         - make
         - verify
-        - WHAT=typecheck
-        - KUBE_JUNIT_REPORT_DIR=${WORKSPACE}/_artifacts
+        command:
+        - runner.sh
```
Comment (stevekuznetsov) [inline config/jobs/kubernetes-security/generated-security-jobs.yaml]: What does the runner do?

Comment (krzyzacy) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: runner gives some hacky things like... GOPATH in PATH, activate service account, enable bazel cache, docker-in-docker, etc, wonder if you need any of those here though :-)

Comment (fejta) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: Why do we need the service account? Bootstrap needs this to do the gcs upload, but this is now handled by other containers

Comment (krzyzacy) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: I'm not exactly sure what the job itself is doing, if it doesn't do things like gcs calls you can get rid of the service account preset as well.

Comment (stevekuznetsov) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: We should do `$GOPATH/bin` in `$PATH` in the decoration

**Pull request merged**